### ISSUE TITLE
Update Coord system and Notation page in docs

### ISSUE THREAD

**Issue by tomchor** [CliMA/Oceananigans.jl#1779]
I think [Coordinate system and notation](https://clima.github.io/OceananigansDocumentation/stable/physics/notation/) page in the docs needs to be updated. 

First, we don't only use a Cartesian grid anymore (we don't even use the word Cartesian for the `RegularRectilinearGrid` anymore). I'm not sure how you guys wanna adapt this page and notation for the spherical coordinate grid though.

Second, a part of the math isn't rendering and I'm not sure why.

**Comment by glwagner** [comment]
Just one clarification --- I think we are still almost always locally Cartesian (the main exception being the lack of a "global" native coordinate system for cubed sphere stuff), because we still describe locations with a 3 tuple (eg we use three orthogonal coordinates). However it's true that on `RegularLatitudeLongitudeGrid` we use longitude, latitude, and altitude as coordinates rather than x, y, z.

With the current software design we probably need to describe the coordinate system for each `AbstractGrid` separately.

**Comment by francispoulin** [comment]
I think this does need some improvement.  Outside of `Oceananigans` I would have said the two grids are `Cartesian` and `Spherical` but I know that there was a long discussion and we settled on `Rectangular`, which makes sense.  

I wonder whether we want to continue to have `RegularLatitudeLongitudeGrid` or perhaps `RegularSphericalGrid`.  It is a bit shorter and maybe more similar to `RegularRectangularGrid`?

**Comment by glwagner** [comment]
@francispoulin is it ok if we limit the discussion in this issue to the documentation and discuss naming for grid types elsewhere?

It looks like I've been confused about the definition of "Cartesian" ever since we changed the name of the grid type from Cartesian to Rectilinear.

I think the important point is that the grid type specifies both a coordinate system (implicitly) and a mesh (via nodal points, grid spacings, areas, volumes, etc).

We don't really have full support for spherical coordinate systems yet, because to support a coordinate system on the cubed sphere we need to have an abstraction for vectors that can translate user input in some global coordinate system to the local coordinate system (different on each facet of the cubed sphere).

**Comment by navidcy** [comment]
I agree: updates are indeed in order but *only* when spherical coordinate grids are fully supported out there.

**Comment by navidcy** [comment]
Perhaps it's time for this ;)

**Comment by glwagner** [comment]
Also, is the label "Physics" the best one for this category?

Side note: "Model setup" should definitely come right after "Examples"...
